src/doghouse/core/services/delta_engine.py

```
@@ -0,0 +1,42 @@
+        current_map: Dict[str, Blocker] = {b.id: b for b in current.blockers}
+        
+        removed_ids = baseline_ids - current_ids
+        added_ids = current_ids - baseline_ids
+        still_open_ids = baseline_ids & current_ids
+        
+        return Delta(
+            baseline_timestamp=baseline.timestamp.isoformat(),
+            current_timestamp=current.timestamp.isoformat(),
+            baseline_sha=baseline.head_sha,
+            current_sha=current.head_sha,
+            added_blockers=[current_map[id] for id in added_ids],
```

Ruff (0.15.7)

[error] Variable `id` is shadowing a Python builtin

(A001)